### PR TITLE
Add ToJson and FromJson template functions

### PR DESCRIPTION
**Pull request by yuvipanda**
Similar to ToYAML, is useful when you need to specify JSON as values for things like environment variables / values of config, etc.

I need to figure out how to get go to build things again on my laptop, and then add tests.

(I've needed this exact thing several times, and someone on #helm asked for this too recently so)

### CODE REVIEW

**Comment by yuvipanda** [comment]
Tests have been added, and I think this is ready to review!

**Comment by ahawkins** [comment]
I like this because it may make it easier to keep init container definitions in `values.yaml`. Is there a way to wrap that output in single quotes?

File: pkg/chartutil/files_test.go [hunk 1]
```diff
@@ -141,3 +141,52 @@ one:
 		t.Fatal("Expected parser error")
 	}
 }
+
+func TestToJson(t *testing.T) {
+	expect := "{\"foo\":\"bar\"}"
```
Comment (bacongobbler) [inline pkg/chartutil/files_test.go]: You can simplify the backslash escapes by using backticks to surround the template. Makes it easier to read :)

```
expect := `{"foo":"bar"}`
```

**Comment by bacongobbler** [comment]
@ahawkins you can wrap anything using `printf`. e.g. `{{ printf "'%s'" (toJson $myvar) }}`. Was that what you're looking for?

**Comment by technosophos** [comment]
@ahawkins There's also `squote`: `toJSON $thing | squote`. I can't remember how much escaping `squote` does, though.

**Comment by technosophos** [comment]
I agree with @bacongobbler 's suggestion, but won't make it a precondition for merging this patch, since it's fully functional as-is. I'll leave the PR open for a bit to give @yuvipanda a chance to fix if desired. Otherwise, I'll just merge.

File: pkg/chartutil/files_test.go (hunk 1 above)
Comment (yuvipanda) [inline pkg/chartutil/files_test.go]: done!

**Comment by yuvipanda** [comment]
@technosophos fixed! Thanks @bacongobbler for the suggestion!

Thanks everyone for the quick response!

**Review by bacongobbler — APPROVED** [review]

**Comment by yuvipanda** [comment]
w00t. What else needs to happen before this gets merged? Excited :D

**Comment by technosophos** [comment]
Thanks!

**Pull request merged**